### DISCUSSION TITLE
[QUESTION] Can I implement real-time drawing sharing(online web drawing game) with fabric.js?

### DISCUSSION THREAD

**Discussion by nayounsang** [fabricjs/fabric.js#9975] [Q&A] [answered]
# Can I implement real-time paint sharing(online web drawing game such as skrbbl) with fabric.js?
## What I want to implement
* The host draws a picture in canvas with fabric.js. 
* Participants can see a drawing drawn by the host in real time.
## Problem
* Fabric.js saves paint datas with Fabric Object.
* Fabric Object is the small drawing, not atomic data.
* For example, in freedrawing mode, it is not saved while the mouse is moving, but is saved after the mouse stops.
* So, If My app share this, users will see a choppy and slow picture.
* This is image for understanding. I captured this image when I was drawing in canvas. But # of object is 0. Because I haven't stoped the mouse yet.
![image](https://github.com/fabricjs/fabric.js/assets/77400131/7a7eb55a-4938-4045-8f59-3dfecb61690f)
* This is code. I use the "mouse:move" event.
```js
  useEffect(() => {
    canvas.on("mouse:move", () => {
      const obj = canvas.toObject();
      console.log(obj);
    });
  }, [canvas]);
```
## Detailed Question
* I want to know valid event for real-time drawing sharing.
* How to I get the data created and updated when I am drawing in canvas.

**Comment by asturur** [comment, marked as answer]
you probably need to build your own custom brush, and use mouse move or the brush render function for it.
This goes deep into application building, and is out of scope for the help i can give you.
